File: backend/src/main/java/com/barbuddy/shared/measurement/Measurement.java
package com.barbuddy.shared.measurement;

import jakarta.persistence.Embeddable;
import java.math.BigDecimal;

/** A reviewed display measurement; US and metric values are stored independently. */
@Embeddable
public class Measurement {
  private BigDecimal quantity;
  private BigDecimal maximumQuantity;
  private String unit;
  private String modifier;

  protected Measurement() {}

  public BigDecimal getQuantity() {
    return quantity;
  }

  public BigDecimal getMaximumQuantity() {
    return maximumQuantity;
  }

  public String getUnit() {
    return unit;
  }

  public String getModifier() {
    return modifier;
  }
}

File: frontend/src/features/catalog/measurement.ts
import type { DisplayMeasurement } from '@/api/generated/models';

export function measurement(value?: DisplayMeasurement) {
  if (!value) return '';
  const quantity =
    value.quantity == null
      ? ''
      : `${value.quantity}${value.maximumQuantity == null ? '' : `–${value.maximumQuantity}`}`;
  return [value.modifier, quantity, value.unit].filter(Boolean).join(' ');
}
